[Lens] [Docs] Add more QA for Lens (#97142)
elastic/kibana: docs/user/dashboard/lens.asciidoc

[role="screenshot"]
image::images/lens_value_labels_xychart_toggle.png[Lens Bar chart value labels menu]

> Note: it is not possible yet to control the labels position in bar visualizations.

Reviewer: ```suggestion
NOTE: In bar charts, you are unable to move the label positions.
```

* *Pie*, *Donut*, and *Treemap*
+
[role="screenshot"]
image::images/lens_value_labels_partition_toggle.png[Lens Pie chart value labels menu]

> Other type of visualizations do not support labels yet.

Reviewer: I don't think we need this sentence. In the intro to the section, we already say `A subset of **Lens** visualizations support value labels.`

[[is-it-possible-to-sort-dimensions-in-a-chart]]
===== How do I sort the dimensions in a chart?

> Sorting dimensions in visualizations is unsupported in *Lens* for charts.

Reviewer: ```suggestion
Sorting chart dimensions is unsupported in *Lens*, but you can sort the dimensions for a single column in data tables.
```

Author: I see the copy is shorter with your version, but it does not sound as clear (to me) as having two distinct sentences.

> Sorting is possible in *Lens* datatables, but it is limited to a single column. Click on the column header to sort by the column and select the sorting criteria.

Reviewer: ```suggestion
To sort the dimension of a single column, click the column header, then select the sorting criteria you want to use.
```

> Sorting is possible in *Lens* datatables, but it is limited to a single column. Click on the dimension header to sort by the dimension/column and select the sorting criteria.

Reviewer 2: ```suggestion
Sorting is possible in *Lens* datatables, but it is limited to a single column. Click on the column header to sort by the column and select the sorting criteria.
```

> Note that if the dimension is used within the `Columns` (transposed) then all columns that belong to the same dimension will be sorted.

Reviewer: ```suggestion
If the dimension is used within the transposed `Columns`, then all columns that belong to the same dimension are sorted.
```

Author: Your version sounds better. I guess the only pending issue is to understand how to call the new `transposed` feature for data tables. I'll check the release notes for some reference.

Reviewer: The definition of transpose is "to change the relative position, order, or sequence of; cause to change places; interchange"

I suggest change this to:

If you use the dimension in the reordered columns, then all columns that belong to the same dimension are sorted.

===== How do I change the number of suggestions?

Configuring the *Suggestions* that *Lens* automatically populates is unsupported.

[float]
[[is-it-possible-to-use-different-indexpatterns-in-lens]]
===== Is it possible to use different index patterns in a single visualization?

> You can create a visualization from multiple index patterns in *Lens*: this feature is available for a subset of visualizations such *Bar*, *Line* and *Area* charts.

Reviewer 2: ```suggestion
You can create a visualization from multiple index patterns in *Lens*: this feature is available for a subset of visualizations such as *Bar*, *Line* and *Area* charts.
```

Each *Layer* in a visualization is associated with an index pattern and mutiple *Layers* can be combined together within the same visualization. Each *Layer* also has a chart switcher button in order to select the best type of visualization for the specific dataset.
At the same time it is possible to dynamically change the indexpattern for the *Layer* when required.

[float]
[[why-my-field-x-is-missing-from-the-fields-list]]

> ===== Why is my field X missing from the fields list in Lens?

Reviewer 2: that's a good one, I saw this like 3 times in Discuss forum 👏🏼

*Lens* does not support the visualization of full-text fields, therefore it is not showing them in the data summary.
You will still see the `X.keyword` field, when available.

[float]
[[how-to-handle-gaps-in-time-series-visualizations]]
===== How to handle gaps in time series visualizations?

*Lens* provides a set of features to handle missing values for *Area* and *Line* visualization.
This is useful for sparse data in time series.

> To select a different way to represent missing values, click on the *brush* button above the visualization in the editor to see a selection of strategies to handle missing values. The default strategy is to hide the gaps.

Reviewer 2: Maybe the image of the brush button would be useful here?

[float]
[[is-it-possible-to-change-the-scale-of-Y-axis]]

> ===== Is it possible to change the scale of the Y axis in a visualization?

Reviewer 2: ```suggestion
===== Is it possible to statically define the scale of the Y axis in a visualization?
```
Nit: I think `change` can be misleading, but I want to hear more opinions about it!

Reviewer 2: Btw Joe might be faster 😁 https://github.com/elastic/kibana/issues/67614

> At the moment it is not possible to change the scale of the Y axis (for instance, making it start from another value than 0) in *Lens*.

Reviewer 2: Nit: Maybe it would be worth mentioning that at the moment we only support linear scale.

Author: Good catch!

===== How do I change the number of suggestions?

Configuring the *Suggestions* that *Lens* automatically populates is unsupported.

[float]
[[is-it-possible-to-use-different-indexpatterns-in-lens]]

> ===== Is it possible to use different index patterns in a single visualization?

Reviewer: ```suggestion
===== Can I visualize multiple index patterns in a single visualization?
```

> You can create a visualization from multiple index patterns in *Lens*: this feature is available for a subset of visualizations such as *Bar*, *Line* and *Area* charts.

Reviewer: ```suggestion
You can create *Bar*, *Line* and *Area* charts from multiple index patterns.
```

> Each *Layer* in a visualization is associated with an index pattern and mutiple *Layers* can be combined together within the same visualization. Each *Layer* also has a chart switcher button in order to select the best type of visualization for the specific dataset.

Reviewer: ```suggestion
Each chart *Layer* is associated with an index pattern, and you can combine multiple *Layers* in the same chart. Each *Layer* also has a menu to change the visualization type, which allows you to select the best type of visualization for the data.
```

Reviewer: Since this section is about using multiple index patterns, is this sentence necessary?

Author: I think that, while it may sounds a distinct issue, there is often a correlation between overlaying different charts types from different index patterns.

> At the same time it is possible to dynamically change the indexpattern for the *Layer* when required.

Reviewer: ```suggestion
You can also change the index pattern for a single *Layer*.
```

[float]
[[why-my-field-x-is-missing-from-the-fields-list]]

> ===== Why is my field X missing from the fields list in Lens?

Reviewer: ```suggestion
===== Why is my field X missing from the fields list?
```

> *Lens* does not support the visualization of full-text fields, therefore it is not showing them in the data summary.

Reviewer: ```suggestion
Creating visualizations with full-text fields is unsupported in *Lens*. When your index pattern contains a full-text field, *Lens* displays the `X.keyword` field, but is unable to display the data summary.
```

Author: I think the suggested version is not correct, as it assumes always the availability of the `X.keyword` field for each full-text field, while this is not true.

Reviewer: How about:

Creating visualizations with full-text fields is unsupported in *Lens*. When your index pattern contains full-text fields, the fields do not appear in the data summaries.

Author: The bits of information to communicate, I think, are the following three:
* *Lens* does not support full-text fields
* If a full-text field has also `keyword` subfield then the latter is shown
* Data summaries are not available for full-text fields, even when the `keyword` subfield is available.

In particular the first 2 are the most important part, while the third could be omitted if we want - it's just nice as it adds an extra bit to the essential information.

> You will still see the `X.keyword` field, when available.

Reviewer: ```suggestion

```

[float]
[[how-to-handle-gaps-in-time-series-visualizations]]

> ===== How to handle gaps in time series visualizations?

Reviewer: ```suggestion
===== How do I handle gaps in time series visualizations?
```

> *Lens* provides a set of features to handle missing values for *Area* and *Line* visualization.

Reviewer: ```suggestion
*Lens* provides a set of features to handle missing values for *Area* and *Line* charts, which is useful for sparse data in time series data.
```

> This is useful for sparse data in time series.

Reviewer: ```suggestion
```

> To select a different way to represent missing values, click on the *brush* button above the visualization in the editor to see a selection of strategies to handle missing values. The default strategy is to hide the gaps.

Reviewer: ```suggestion
To select a different way to represent missing values, open the *Visual options* menu, then select how to handle missing values. The default is to hide the missing values.
```

+
[role="screenshot"]
image::images/lens_missing_values_strategy.png[Lens Missing values strategies menu]

[float]
[[is-it-possible-to-change-the-scale-of-Y-axis]]

> ===== Is it possible to statically define the scale of the Y axis in a visualization?

Reviewer: ```suggestion
===== Is it possible to statically define the scale of the y-axis in a visualization?
```

> At the moment it is not possible to change the scale of the Y axis (for instance, making it start from another value than 0, or use a logarithmic scale.) in *Lens*.

Reviewer: ```suggestion
The ability to start the y-axis from another value than 0, or use a logarithmic scale, is unsupported in *Lens*.
```

[float]
[[is-it-possible-to-have-pagination-for-datatable]]

> ===== Is it possible to have pagination in the datatable visualization?

Reviewer: ```suggestion
===== Is it possible to have pagination in a data table?
```

> At the moment it is not possible to have pagination on a *Lens* datatable. However the <<types-of-visualizations,aggregation based data table>> does support this feature.

Reviewer: ```suggestion
Pagination in a data table is unsupported in *Lens*. However, the <<types-of-visualizations,aggregation-based data table>> supports pagination.
```

[float]
[[is-it-possible-to-have-more-than-one-Y-axis-scale]]

> ===== Is it possible to have more than one Y axis scale in visualization?

Reviewer: ```suggestion
===== Is it possible to have more than one y-axis scale in visualizations?
```

> *Lens* lets you pick, for each Y dimension, up to two distinct axis: *left* and *right*. Each axis can have a different scale.

Reviewer: ```suggestion
For each y-axis in the editor, *Lens* supports *left* and *right* axes. Each axis can have a different scale.
```

Author: > For each y-axis in the editor

It does not sound not to immediate to me exactly what it meant.

===== How do I change the number of suggestions?

Configuring the *Suggestions* that *Lens* automatically populates is unsupported.

[float]
[[is-it-possible-to-use-different-indexpatterns-in-lens]]
===== Can I visualize multiple index patterns in a single visualization?

You can create *Bar*, *Line* and *Area* charts from multiple index patterns.

Each *Layer* in a visualization is associated with an index pattern and mutiple *Layers* can be combined together within the same visualization. Each *Layer* also has a chart switcher button in order to select the best type of visualization for the specific dataset.
You can also change the index pattern for a single *Layer*.

[float]
[[why-my-field-x-is-missing-from-the-fields-list]]

> ===== Why is my field X missing from the fields list?

Reviewer 3: There are a few possible reasons for a field to be missing:

* Full text
* Geo (in all released versions)
* Mapping is not supported by index patterns, like `flattened` or `object`

Also, sometimes user say "missing" when it's actually in the "empty" list. Would be nice to summarize all of these if it can be done concisely, but I don't think it's a blocker.

Author: Makes sense. I'll merge this and revisit that particular section in a subsequent PR
